### ISSUE TITLE
[Python] Implement Scanner::from_fragments() to initialize a scanner from sub-set of fragments

### ISSUE THREAD

**Issue by eddyxu** [eto-ai/lance#860]
# Problem Statement

It will be useful for fine-tuned distributed scan. a `master` node can decide the plan that assigns fragments to each distributed node. Each distributed worker could scan the fraction of the dataset accordingly.

**Comment by GallagherCommaJack** [comment]
ideally the API for this accepts an iterator of fragments (vs a list)

**Comment by changhiskhan** [comment]
This means adding a fragment_ids setter or smth to Scanner and modify the Scan IO node to take this list as a parameter ?

**Comment by eddyxu** [comment]
Yes, this could be just a constructor from `Scanner`, takes in the fragments, and build a scanner in one worker. 

Can follow the pattern like https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html#pyarrow.dataset.Scanner.from_fragment
